org.eclipse.m2e.core/src/org/eclipse/m2e/core/internal/MavenPluginActivator.java

```diff
@@ -263,6 +266,12 @@ public void start(final BundleContext context) throws Exception {
     bundleContext.registerService(IMavenProjectRegistry.class, projectManager, null);
   }
 
+  private static Dictionary<String, ?> getMaskProperties(int mask) {
+    Dictionary<String, Object> properties = new Hashtable<>();
+    properties.put("event.mask", mask);
+    return properties;
```

Comment:
In case you don't want to create a Hashtable you could use
```suggestion
    return FrameworkUtil.asDictionary(Map.of("event.mask", mask));
```

Reply:
I don't know if wrapping things more really makes things more easy to understand but nice to know in case one has already a Map ;-)

Reply:
That's right. IIRC a method for a singleton Dictonary exists somehwere but I did not find it quickly.
Any way I just wanted to mention it.

Reply:
We hoepfully could remove that method on the next eclipse release.